extensions/test-src/org/pentaho/platform/web/http/api/resources/services/UserRoleDaoServiceTest.java

```diff
@@ -523,4 +526,39 @@ public void testGetRoleBindingStructSecurityException() {
       return returnedValues.equals( correctValues );
     }
   }
+  
+  private boolean validationFailed( UserRoleDaoService service, User user ) {
+    try {
+      service.createUser( user );
+    } catch ( UserRoleDaoService.ValidationFailedException e ) {
+      return true;
+    } catch ( Exception e ) {
+      return false;
+    }
+    return false;
+  }
+
+  @Test
+  public void testCreateUserValidation() {
+    List<String> reservedChars = Arrays.asList( "/,\\,\t,\r,\n".split( "," ) );
+    IUnifiedRepository repo = mock( IUnifiedRepository.class );
+    doReturn( reservedChars ).when( repo ).getReservedChars();
+    PentahoSystem.registerObject( repo );
+    UserRoleDaoService service = new UserRoleDaoService(){
```

Comment:
See my unit test examples, we should not be mocking this, the code in private methods should be exercised through the call to the method that accesses them. Mock the objects that PentahoSystem.get() returns.